collections/aggregate_groups.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Applies the given aggregator to each group in the given Grouping, returning the results together with the respective group keys
+ *
+ * ```ts
+ * import { aggregateGroups } from "./aggregate_groups.ts"
+ * import { assertEquals } from "../testing/asserts.ts";
+ * 
+ * const foodProperties = {
+ *     'Curry': [ 'spicy', 'vegan' ],
+ *     'Omelette': [ 'creamy', 'vegetarian' ],
+ * }
+ * const descriptions = aggregateGroups(foodProperties,
+ *     (acc, current, key, first) => {
+ *         if (first)
+ *             return `${key} is ${current}`
+ *
+ *         return `${acc} and ${current}`
+ *     },
+ *     '',
+ * )
+ *
+ * assetEquals(descriptions === {
+ *     'Curry': 'Curry is spicy and vegan',
+ *     'Omelette': 'Omelette is creamy and vegetarian',
+ * })
+ * ```
+ */
+export function aggregateGroups<T, A>(
+  record: Record<string, Array<T>>,
+  aggregator: (accumulator: A, current: T, key: string, first: boolean) => A,
+  initalValue: A,
```

Comment:
I think we actually don't need this with the aggregator signature

Reply:
I don't get it. if we don't want to accept the initial value as an argument what should we pass to the reducer as the initial value?

Reply:
The aggregator gets the first element (and first = true) for the first call